smithy-aws-protocol-tests/model/awsJson1_0/errors.smithy

```diff
@@ -58,6 +58,23 @@ apply InvalidGreeting @httpResponseTests([
               }""",
         bodyMediaType: "application/json",
     },
+    {
+        id: "AwsJson10FooErrorWithMultipleAmznErrorTypes",
+        documentation: """
+            API Gateway always adds its own x-amzn-errortype header on gateway responses.
+            However, this header does not match the error configured in customers model and \
+            results in two X-Amzn-Errortype header values. \
+            The first value is provided by the customer and the second value is from API Gateway \
+
+            Clients need to split the header value on ',' and take only the first element. \
+            For example, 'InvalidGreeting,InvalidParameterException' is to be interpreted as 'InvalidGreeting'.""",
+        protocol: awsJson1_0,
+        code: 400,
+        headers: {
+            "X-Amzn-Errortype": "InvalidGreeting,InvalidParameterException",
```

Comment:
Can this be applied on `FooError` instead, where the `X-Amzn-Errortype` behavior is currently expressed?